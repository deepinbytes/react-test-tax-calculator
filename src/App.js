import React, {Component} from "react";
import "./App.css";
import Appbar from "./Components/Appbar";
import Footer from "./Components/Footer";
import TaxBracket from "./Components/TaxBracket";
import Container from "@material-ui/core/Container";
import TextField from "@material-ui/core/TextField";
import Typography from "@material-ui/core/Typography";
import VisibilityIcon from "@material-ui/icons/Visibility";
import VisibilityOffIcon from "@material-ui/icons/VisibilityOff";
import Fab from "@material-ui/core/Fab";
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormControl from '@material-ui/core/FormControl';
import FormLabel from '@material-ui/core/FormLabel';

const RELIEF_PROFILE_TK0 = 54000, RELIEF_PROFILE_K0 = 58500, RELIEF_PROFILE_K1 = 63000, RELIEF_PROFILE_K2 = 67500,
    RELIEF_PROFILE_K3 = 72000;

class App extends Component {

    constructor(props) {
        super(props);

        this.state = {
            value: '',
            monthlyIncome: "",
            monthlyTax: "",
            salaryAfterTax: "",
            annualIncome: "",
            annualTax: "",
            annualIncomeAfterTax: "",
            taxBracketStatus: false,
            limit: false,
            profile: "TK0",
            annualIncomeAfterTaxRelief: "",
        };

        this.processDetails = this.processDetails.bind(this);

    }

    currencyFormat = num => num.replace(/(\d)(?=(\d{3})+(?!\d))/g, "$1.") + " IDR";

    setDetails = (value, limit, monthlyIncome, monthlyTax, salaryAfterTax,
                  annualIncome, annualTax, annualIncomeAfterTax, annualTaxAfterRelief) => {
        this.setState({
            value: value,
            limit: limit,
            monthlyIncome: this.currencyFormat(parseFloat(monthlyIncome).toFixed(3)),
            monthlyTax: this.currencyFormat(parseFloat(monthlyTax).toFixed(3)),
            salaryAfterTax: this.currencyFormat(parseFloat(salaryAfterTax).toFixed(3)),
            annualIncome: this.currencyFormat(parseFloat(annualIncome).toFixed(3)),
            annualTax: this.currencyFormat(parseFloat(annualTax).toFixed(3)),
            annualIncomeAfterTax: this.currencyFormat(parseFloat(annualIncomeAfterTax).toFixed(3)),
            annualIncomeAfterTaxRelief: this.currencyFormat(parseFloat(annualTaxAfterRelief).toFixed(3))
        });
    };


    calculateTaxRelief = annualSalary => {
        /*
        Tax Relief
        TK0 = 54.000.000 IDR
        K0 = 58.500.000 IDR
        K1 = 63.000.000 IDR
        K2 = 67.500.000 IDR
        K3 = 72.000.000 IDR
        */
        let profile = this.state.profile;
        if (profile === "TK0") {
            return annualSalary - RELIEF_PROFILE_TK0
        }
        if (profile === "K0") {
            return annualSalary - RELIEF_PROFILE_K0
        }
        if (profile === "K1") {
            return annualSalary - RELIEF_PROFILE_K1
        }
        if (profile === "K2") {
            return annualSalary - RELIEF_PROFILE_K2
        }
        if (profile === "K3") {
            return annualSalary - RELIEF_PROFILE_K3
        }
    };

    computeTax = income => {
        /*
        Annual Income from 0 to 50.000.000 IDR - tax rate is 5 %.
        Annual Income from 50.000.000 to 250.000.000 IDR - tax rate is 15 %.
        Annual Income from 250.000.000 to 500.000.000 IDR - tax rate is 25 %.
        Annual Income above 500.000.000 IDR - tax rate is 30 %.
        */
        const tierOneRate = 0.05;
        const tierTwoRate = 0.15;
        const tierThreeRate = 0.25;
        const tierFourRate = 0.30;

        const bracketOne = 50000;
        const bracketTwo = 250000;
        const bracketThree = 500000;

        const bracketOneMaxTax = tierOneRate * bracketOne;
        const bracketTwoMaxTax =
            bracketOneMaxTax + tierTwoRate * (bracketTwo - bracketOne);
        const bracketThreeMaxTax =
            bracketTwoMaxTax + tierThreeRate * (bracketThree - bracketTwo);


        if (income <= bracketOne) return income * tierOneRate;
        if (income > bracketOne && income <= bracketTwo)
            return bracketOneMaxTax + tierTwoRate * (income - bracketOne);
        if (income > bracketTwo && income <= bracketThree)
            return bracketTwoMaxTax + tierThreeRate * (income - bracketTwo);
        return bracketThreeMaxTax + tierFourRate * (income - bracketThree);
    };

    recalculate = () => {
        let income = this.state.value;
        let annualIncome = income * 12;
        let annualIncomeAfterTaxRelief = this.calculateTaxRelief(annualIncome);
        let annualTax = this.computeTax(annualIncomeAfterTaxRelief);
        let monthlyTax = annualTax / 12;
        let annualIncomeAfterTaxDeduction = annualIncome - annualTax;
        let takeHomePay = annualIncomeAfterTaxDeduction / 12;

        this.setDetails(
            income,
            false,
            income,
            monthlyTax,
            takeHomePay,
            annualIncome,
            annualTax,
            annualIncomeAfterTaxDeduction,
            annualIncomeAfterTaxRelief
        );
        if (
            income === "" ||
            income === 0 ||
            income === null ||
            income === undefined
        ) {
            this.setState({
                monthlyIncome: "",
                monthlyTax: "",
                salaryAfterTax: "",
                annualIncome: "",
                annualTax: "",
                annualIncomeAfterTax: "",
                limit: false,
                value: '',
                annualIncomeAfterTaxRelief: ''
            });
        }
    };

    processDetails = e => {
        const income = Number(e.target.value);
        let flag = false;

        const re = /^[0-9\b]+$/;
        if (e.target.value === '' || re.test(e.target.value)) {
            flag = true;
        }
        let annualIncome = income * 12;
        let annualIncomeAfterTaxRelief = this.calculateTaxRelief(annualIncome);
        let annualTax = this.computeTax(annualIncomeAfterTaxRelief);
        let monthlyTax = annualTax / 12;
        let annualIncomeAfterTaxDeduction = annualIncome - annualTax;
        let takeHomePay = annualIncomeAfterTaxDeduction / 12;

        if (flag) {
            this.setDetails(
                income,
                false,
                income,
                monthlyTax,
                takeHomePay,
                annualIncome,
                annualTax,
                annualIncomeAfterTaxDeduction,
                annualIncomeAfterTaxRelief
            );
        }
        if (
            income === "" ||
            income === 0 ||
            income === null ||
            income === undefined
        ) {
            this.setState({
                monthlyIncome: "",
                monthlyTax: "",
                salaryAfterTax: "",
                annualIncome: "",
                annualTax: "",
                annualIncomeAfterTax: "",
                limit: false,
                value: '',
            });
        }
    };


    handleProfileChange = (event) => {
        this.setState({
            profile: event.target.value,
        });
        this.recalculate()
    };


    render() {
        const {
            monthlyIncome,
            monthlyTax,
            salaryAfterTax,
            annualIncome,
            annualTax,
            annualIncomeAfterTax,
            annualIncomeAfterTaxRelief,
            profile,
            taxBracketStatus,
            limit
        } = this.state;


        return (
            <div>
                <Appbar/>
                <Container>
                    <br/>
                    <Typography variant="h5">
                        Enter Your Monthly Salary
                    </Typography>

                    <TextField
                        id="monthly-sal"
                        label="Monthly Salary "
                        type="text"
                        margin="normal"
                        variant="outlined"
                        value={this.state.value}
                        fullWidth={true}
                        inputProps={{
                            style: {fontSize: 30},
                            maxLength: "12",
                        }}
                        onChange={this.processDetails}
                    />
                    <FormControl component="fieldset">
                        <FormLabel component="legend">Profile</FormLabel>
                        <RadioGroup aria-label="profile" name="profile" value={this.state.profile}
                                    onChange={this.handleProfileChange}>
                            <FormControlLabel value="TK0" control={<Radio/>} label="Single"/>
                            <FormControlLabel value="K0" control={<Radio/>} label="Married with no dependants"/>
                            <FormControlLabel value="K1" control={<Radio/>} label="Married with 1 dependants"/>
                            <FormControlLabel value="K2" control={<Radio/>} label="Married with 2 dependants"/>
                            <FormControlLabel value="K3" control={<Radio/>} label="Married with 3 dependants"/>
                        </RadioGroup>
                    </FormControl>
                    <br/>
                    <br/>

                    {limit && (
                        <div>
                            <Typography variant="h3">
                                Invalid Input.
                            </Typography>
                        </div>
                    )}
                </Container>

                {!limit && (
                    <div className="container">
                        <div>
                            <div className="row">
                                <div className="col-25">
                                    <label>Monthly Income</label>
                                </div>
                                <div className="col-75">
                                    <input id="monthly-income" type="text" value={monthlyIncome} disabled/>
                                </div>
                            </div>


                            <div className="row">
                                <div className="col-25">
                                    <label>Annual Income</label>
                                </div>
                                <div className="col-75">
                                    <input id="annual-income" type="text" value={annualIncome} disabled/>
                                </div>
                            </div>


                            <div className="row">
                                <div className="col-25">
                                    <label>Tax Relief Profile</label>
                                </div>
                                <div className="col-75">
                                    <input type="text" value={profile} disabled/>
                                </div>
                            </div>
                            <div className="row">
                                <div className="col-25">
                                    <label>Annual Taxable Income</label>
                                </div>
                                <div className="col-75">
                                    <input id="annual-taxable-income" type="text"
                                           style={{color: "red", fontWeight: "600"}}
                                           value={annualIncomeAfterTaxRelief} disabled/>
                                </div>
                            </div>

                            <div className="row">
                                <div className="col-25">
                                    <label>Annual Tax Income</label>
                                </div>
                                <div className="col-75">
                                    <input id="annual-tax-income" type="text" style={{color: "red", fontWeight: "600"}}
                                           value={annualTax}
                                           disabled/>
                                </div>

                            </div>
                            <div className="row">
                                <div className="col-25">
                                    <label>Annual Income After Deductions</label>
                                </div>
                                <div className="col-75">
                                    <input
                                        id="annual-income-deductions"
                                        type="text"
                                        value={annualIncomeAfterTax}
                                        style={{color: "green", fontWeight: "600"}}
                                        disabled
                                    />
                                </div>
                            </div>
                            <div className="row">
                                <div className="col-25">
                                    <label>Monthly Tax</label>
                                </div>
                                <div className="col-75">
                                    <input
                                        id="monthly-tax"
                                        type="text"
                                        value={monthlyTax}
                                        style={{color: "red", fontWeight: "600"}}
                                        disabled
                                    />
                                </div>
                            </div>
                            <div className="row">
                                <div className="col-25">
                                    <label>Take Home Pay(Monthly)</label>
                                </div>
                                <div className="col-75">
                                    <input id="take-home" type="text" style={{color: "green", fontWeight: "600"}}
                                           value={salaryAfterTax} disabled/>
                                </div>
                            </div>
                        </div>
                    </div>

                )}
                <br/>

                {!taxBracketStatus && (
                    <div align="center"><Fab
                        id = "tax-bracket"
                        color="primary"
                        variant="extended"
                        aria-label="Delete"
                        style={{margin: "0", marginTop: "5px", backgroundColor: "black"}}
                        onClick={() => {
                            this.setState({taxBracketStatus: !taxBracketStatus});
                        }}
                    >
                        <VisibilityIcon style={{marginRight: "0"}}/>
                        ­Show Income Tax Bracket
                    </Fab></div>
                )}
                <br/>
                {taxBracketStatus && (
                    <Container>
                        <TaxBracket/>
                        <div align="center"><Fab
                            id = "tax-bracket-close"
                            color="secondary"
                            variant="extended"
                            aria-label="Delete"
                            style={{margin: "0", backgroundColor: "black"}}
                            onClick={() => {
                                this.setState({taxBracketStatus: !taxBracketStatus});
                            }}
                        >
                            <VisibilityOffIcon style={{marginRight: "0"}}/>
                            ­Hide Income Tax Bracket
                        </Fab></div>
                    </Container>
                )}
                <br/>

                <Footer/>
            </div>
        );
    }
}

export default App;