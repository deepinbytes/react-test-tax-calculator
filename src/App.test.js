import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});
let wrapper;

beforeEach(() => {
    wrapper = mount(<App/>);
});

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App/>, div);
    ReactDOM.unmountComponentAtNode(div);
});

describe('App', () => {
    test('snapshot renders', () => {
        const component = renderer.create(<App/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});

describe('state interactions', () => {
    test('should change the tax detail states when onChange function of the monthly salary input is invoked', () => {
        const input = wrapper.find("input").at(0);
        input.simulate('change',
            {target: {value: '500000'}}
        );
        expect(wrapper.state('monthlyIncome')).toEqual('500.000.000 IDR');
        expect(wrapper.state('annualIncome')).toEqual('6.000.000.000 IDR');
        expect(wrapper.state('annualIncomeAfterTax')).toEqual('4.271.200.000 IDR');
        expect(wrapper.state('annualTax')).toEqual('1.728.800.000 IDR');
        expect(wrapper.state('salaryAfterTax')).toEqual('355.933.333 IDR');
        expect(wrapper.state('annualIncomeAfterTaxRelief')).toEqual('5.946.000.000 IDR');
        expect(wrapper.state('profile')).toEqual('TK0');

        input.simulate('change',
            {target: {value: '6500'}}
        );
        expect(wrapper.state('monthlyIncome')).toEqual('6.500.000 IDR');
        expect(wrapper.state('annualIncome')).toEqual('78.000.000 IDR');
        expect(wrapper.state('annualIncomeAfterTax')).toEqual('76.800.000 IDR');
        expect(wrapper.state('annualTax')).toEqual('1.200.000 IDR');
        expect(wrapper.state('salaryAfterTax')).toEqual('6.400.000 IDR');
        expect(wrapper.state('annualIncomeAfterTaxRelief')).toEqual('24.000.000 IDR');
        expect(wrapper.state('profile')).toEqual('TK0');
    });

  test('should change the taxbracket state when tax bracket button is invoked', () => {
    const input = wrapper.find("button");
    expect(wrapper.state('taxBracketStatus')).toEqual(false);
    input.props().onClick();
    expect(wrapper.state('taxBracketStatus')).toEqual(true);
  });

  test('should change the profile state when radio buttons are clicked', () => {
    const input = wrapper.find("input").at(1);
    input.simulate('change',
        {target: {value: 'K1'}}
    );
    const mockedHandleRecalculateTax = jest.fn();
    wrapper.instance().recalculate =  mockedHandleRecalculateTax;
    expect(wrapper.state('profile')).toEqual('K1');
    input.simulate('change',
        {target: {value: 'K2'}}
    );
    expect(wrapper.state('profile')).toEqual('K2');
    input.simulate('change',
        {target: {value: 'K3'}}
    );
    expect(wrapper.state('profile')).toEqual('K3');
    input.simulate('change',
        {target: {value: 'TK0'}}
    );
    expect(wrapper.state('profile')).toEqual('TK0');
    expect(mockedHandleRecalculateTax).toHaveBeenCalledTimes(3);
  });


});

describe('detail display', () => {
    test('should change the tax detail fields when monthly salary input is changed', () => {
        const input = wrapper.find("input").at(0);

        input.simulate('change',
            {target: {value: '500000'}}
        );
        const monthlyIncome = wrapper.find("#monthly-income");
        const annualIncome = wrapper.find("#annual-income");
        const annualTaxIncome = wrapper.find("#annual-tax-income");
        const annualIncomeDeductions = wrapper.find("#annual-income-deductions");
        const monthlyTax = wrapper.find("#monthly-tax");
        const takeHome = wrapper.find("#take-home");
        const annualTaxableincome = wrapper.find("#annual-taxable-income");

        expect(annualTaxableincome.props().value).toEqual('5.946.000.000 IDR');
        expect(monthlyIncome.props().value).toEqual('500.000.000 IDR');
        expect(annualIncome.props().value).toEqual('6.000.000.000 IDR');
        expect(annualTaxIncome.props().value).toEqual('1.728.800.000 IDR');
        expect(annualIncomeDeductions.props().value).toEqual('4.271.200.000 IDR');
        expect(monthlyTax.props().value).toEqual('144.066.667 IDR');
        expect(takeHome.props().value).toEqual('355.933.333 IDR');
    });

});
