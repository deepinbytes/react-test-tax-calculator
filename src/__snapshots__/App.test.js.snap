// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div>
  <div
    className="navbar"
  >
    <p>
      Tax Calculator
    </p>
  </div>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <br />
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      Enter Your Monthly Salary
    </h5>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
        htmlFor="monthly-sal"
        id="monthly-sal-label"
      >
        Monthly Salary 
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="monthly-sal"
          maxLength="12"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          style={
            Object {
              "fontSize": 30,
            }
          }
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              Monthly Salary 
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <fieldset
      className="MuiFormControl-root"
    >
      <legend
        className="MuiFormLabel-root"
      >
        Profile
      </legend>
      <div
        aria-label="profile"
        className="MuiFormGroup-root"
        role="radiogroup"
      >
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-6 Mui-checked MuiIconButton-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={true}
                className="PrivateSwitchBase-input-8"
                disabled={false}
                name="profile"
                onChange={[Function]}
                type="radio"
                value="TK0"
              />
              <div
                className="PrivateRadioButtonIcon-root-9 PrivateRadioButtonIcon-checked-11"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-10"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Single
          </span>
        </label>
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input-8"
                disabled={false}
                name="profile"
                onChange={[Function]}
                type="radio"
                value="K0"
              />
              <div
                className="PrivateRadioButtonIcon-root-9"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-10"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Married with no dependants
          </span>
        </label>
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input-8"
                disabled={false}
                name="profile"
                onChange={[Function]}
                type="radio"
                value="K1"
              />
              <div
                className="PrivateRadioButtonIcon-root-9"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-10"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Married with 1 dependants
          </span>
        </label>
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input-8"
                disabled={false}
                name="profile"
                onChange={[Function]}
                type="radio"
                value="K2"
              />
              <div
                className="PrivateRadioButtonIcon-root-9"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-10"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Married with 2 dependants
          </span>
        </label>
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input-8"
                disabled={false}
                name="profile"
                onChange={[Function]}
                type="radio"
                value="K3"
              />
              <div
                className="PrivateRadioButtonIcon-root-9"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-10"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Married with 3 dependants
          </span>
        </label>
      </div>
    </fieldset>
    <br />
    <br />
  </div>
  <div
    className="container"
  >
    <div>
      <div
        className="row"
      >
        <div
          className="col-25"
        >
          <label>
            Monthly Income
          </label>
        </div>
        <div
          className="col-75"
        >
          <input
            disabled={true}
            id="monthly-income"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-25"
        >
          <label>
            Annual Income
          </label>
        </div>
        <div
          className="col-75"
        >
          <input
            disabled={true}
            id="annual-income"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-25"
        >
          <label>
            Tax Relief Profile
          </label>
        </div>
        <div
          className="col-75"
        >
          <input
            disabled={true}
            type="text"
            value="TK0"
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-25"
        >
          <label>
            Annual Taxable Income
          </label>
        </div>
        <div
          className="col-75"
        >
          <input
            disabled={true}
            id="annual-taxable-income"
            style={
              Object {
                "color": "red",
                "fontWeight": "600",
              }
            }
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-25"
        >
          <label>
            Annual Tax Income
          </label>
        </div>
        <div
          className="col-75"
        >
          <input
            disabled={true}
            id="annual-tax-income"
            style={
              Object {
                "color": "red",
                "fontWeight": "600",
              }
            }
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-25"
        >
          <label>
            Annual Income After Deductions
          </label>
        </div>
        <div
          className="col-75"
        >
          <input
            disabled={true}
            id="annual-income-deductions"
            style={
              Object {
                "color": "green",
                "fontWeight": "600",
              }
            }
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-25"
        >
          <label>
            Monthly Tax
          </label>
        </div>
        <div
          className="col-75"
        >
          <input
            disabled={true}
            id="monthly-tax"
            style={
              Object {
                "color": "red",
                "fontWeight": "600",
              }
            }
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-25"
        >
          <label>
            Take Home Pay(Monthly)
          </label>
        </div>
        <div
          className="col-75"
        >
          <input
            disabled={true}
            id="take-home"
            style={
              Object {
                "color": "green",
                "fontWeight": "600",
              }
            }
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <br />
  <div
    align="center"
  >
    <button
      aria-label="Delete"
      className="MuiButtonBase-root MuiFab-root MuiFab-extended MuiFab-primary"
      disabled={false}
      id="tax-bracket"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "margin": "0",
          "marginTop": "5px",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiFab-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          style={
            Object {
              "marginRight": "0",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          />
        </svg>
        ­Show Income Tax Bracket
      </span>
    </button>
  </div>
  <br />
  <br />
  <div
    className="footer"
  >
    <p>
      Developed by
       
      <a
        href="https://github.com/deepinbytes"
        rel="noopener noreferrer"
        target="_blank"
      >
        Ajay P
      </a>
    </p>
  </div>
</div>
`;
